src/compiler/checker.ts

```diff
@@ -19121,7 +19106,17 @@ namespace ts {
             if (returnType.flags & TypeFlags.ESSymbolLike && isSymbolOrSymbolForCall(node)) {
                 return getESSymbolLikeTypeForNode(walkUpParenthesizedExpressions(node.parent));
             }
-            return returnType;
+            let jsAssignmentType: Type | undefined;
+            if (isInJavaScriptFile(node)) {
+                const decl = getDeclarationOfJSInitializer(node);
+                if (decl) {
+                    const jsSymbol = getMergedSymbol(decl.symbol);

```

Comment:
Use `getSymbolOfNode(decl)` - unless you're intentionally avoiding the late bound node check for some reason.

Reply:
No, I just forgot again. Thanks.